### PR TITLE
Feature: Add printing manager on printing

### PR DESCRIPTION
**Pull request by jane-rose**

### CODE REVIEW

File: src/app/constants/index.js [hunk 1]
```diff
@@ -53,7 +53,8 @@ export const STAGES_3DP = {
     modelLoaded: 11,
     gcodeRendered: 12
 };
-
+export const PRITING_MANAGER_TYPE_MATERIAL = 'material';
```
Comment (parachvte) [inline src/app/constants/index.js:56]: Typo: printing

File: src/app/flux/printing/index.js [hunk 2]
```diff
@@ -416,35 +427,78 @@ export const actions = {
 
         return createdDefinition;
     },
+    updateDefinitionsForManager: (definitionId, type) => async (dispatch, getState) => {
+        const state = getState().printing;
+        const savedDefinition = await definitionManager.getDefinition(definitionId);
```
Comment (parachvte) [inline src/app/flux/printing/index.js:432]: What if savedDefinition not found?

File: src/app/flux/printing/index.js [hunk 3]
```diff
@@ -416,35 +427,78 @@ export const actions = {
...
+        if (type === PRITING_MANAGER_TYPE_MATERIAL) {
+            const newDefinitions = state.materialDefinitions.map((item) => {
+                if (item.definitionId === savedDefinition.definitionId) {
```
Comment (parachvte) [inline src/app/flux/printing/index.js:435]: You can use definitionId directly

File: src/app/flux/widget/WidgetState.js [hunk 4]
```diff
@@ -25,7 +25,7 @@ const DEFAULT_STATE = {
     },
     '3dp': {
         default: {
-            widgets: ['3dp-object-list', '3dp-material', '3dp-configurations', '3dp-output']
+            widgets: ['3dp-object-list', '3dp-material', '3dp-manager', '3dp-configurations', '3dp-output']
```
Comment (parachvte) [inline src/app/flux/widget/WidgetState.js:28]: Use a better name

File: src/app/widgets/PrintingMaterial/styles.styl [hunk 5]
```diff
@@ -1,21 +1,19 @@
...
-    line-height: 1.4
-
-    &.selected
-        border: 2px solid $dark-blue-color
-
-.parameter-table
-    .checkbox
-        height: 26px
-        margin-left: 76px
+.material-select{
```
Comment (parachvte) [inline src/app/widgets/PrintingMaterial/styles.styl:2]: Space

File: src/app/flux/printing/index.js (hunk 2 above)
Comment (jane-rose) [reply src/app/flux/printing/index.js:432]: No way to find no result when updating existing definition

File: src/app/flux/printing/index.js (hunk 3 above)
Comment (jane-rose) [reply src/app/flux/printing/index.js:435]: yep

**Pull request merged**